datafusion/sql/src/expr/arrow_cast.rs

```diff
@@ -167,6 +167,34 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Parses the next timezone
+    fn parse_timezone(&mut self, context: &str) -> Result<Option<String>> {
+        match self.next_token()? {
+            Token::None => Ok(None),
+            Token::Some => {
+                self.expect_token(Token::LParen)?;
+                let timezone = self.parse_double_quoted_string("Timezone")?;
+                self.expect_token(Token::RParen)?;
+                Ok(Some(timezone))
+            }
+            tok => Err(make_error(
+                self.val,
+                &format!("finding Timezone for {context}, got {tok}"),
+            )),
+        }
+    }
```

Comment:
add a parser to parse timezone `Option<String>`
either `None`, or `Some("SomeTimezone")`